### PR TITLE
[WIP] Added autofill functionality

### PR DESCRIPTION
**Pull request by ganttastic**
This PR adds autofill functionality to the engine.

iOS - functionally complete, but has a bug where autofill only shows for non-default keyboards.
Android - still needs work and guidance. Current work could be removed if only iOS implementation is wanted in this PR.
Web - not started

### CODE REVIEW

**Comment by ganttastic** [comment]
@googlebot I signed it!

File: shell/platform/android/io/flutter/embedding/engine/systemchannels/TextInputChannel.java [hunk 1]
```diff
@@ -406,6 +411,95 @@ static TextInputType fromValue(@NonNull String encodedName) throws NoSuchFieldEx
     }
   }
 
+  /**
+   * Types of text content for auto-full functionality
```
Comment (jcivelli-google) [inline shell/platform/android/io/flutter/embedding/engine/systemchannels/TextInputChannel.java]: Nit: autofill

Comment (ganttastic) [reply shell/platform/android/io/flutter/embedding/engine/systemchannels/TextInputChannel.java]: 👍 Fixed.

File: shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java [hunk 2]
```diff
@@ -230,6 +230,12 @@ public InputConnection createInputConnection(View view, EditorInfo outAttrs) {
             outAttrs.actionId = enterAction;
         }
         outAttrs.imeOptions |= enterAction;
+        
+        String autofillHintValue = configuration.textContentType.rawValue;
+        if (autofillHintValue != null) {
+          String[] autofillHints = {autofillHintValue};
+          view.setAutofillHints(autofillHints);
```
Comment (ganttastic) [inline shell/platform/android/io/flutter/plugin/editing/TextInputPlugin.java]: Unsure if this is the correct view to set the auto fill hints on.

**Comment by isaiahtaylor** [comment]
Hello, thanks for your work on this! I'd like to cast my vote in favor of getting this accepted with the iOS capability only and opening a separate PR for the other channels. Unless there is a near timeline for the whole thing being complete. Thank you!

**Comment by ganttastic** [comment]
Full autofill implementation is in PR #17493.

**Pull request closed without merging**